sandbox/grommet-app/src/themes/theme.jsx

```diff
@@ -1163,34 +1186,61 @@ const buildTheme = (tokens, flags) => {
       },
       hover: {
         container: {
-          elevation: 'large',
+          elevation: 'medium',
         },
       },
     },
     checkBox: {
       hover: {
         border: {
+          // applies directly to control (checkbox and toggle switch)
+          // TO DO remove from applying to switch
           color: components.hpe.checkbox.control.hover.borderColor,
-          width: components.hpe.checkbox.control.hover.borderWidth,
+          width:
+            dimensions.borderSize[
+              components.hpe.checkbox.medium.control.borderWidth
+            ] || components.hpe.checkbox.medium.control.borderWidth,
         },
+        // applies to container around control and label
         background: {
-          color: 'transparent',
+          color: 'background-hover',
         },
         // HPE Design System guidance states that pad="none" should be applied on CheckBox
         // when its used outside of a FormField. We will apply this hover treatment in
         // those instances.
-        extend: ({ disabled, pad, theme, toggle }) => css`
-          ${!disabled &&
-          pad === 'none' &&
-          !toggle &&
-          `border: 2px solid ${
-            theme.global.colors[
-              components.hpe.checkbox.control.hover.borderColor
-            ][theme.dark ? 'dark' : 'light']
-          };`}
-        `, // Q: missing token for hover borderWidth? this falls into similar boat as secondary button
-      },
-      color: components.hpe.switch.control.handle.rest.background,
+        extend: ({ disabled, pad, theme, toggle, checked }) => {
+          let borderColor;
+          if (toggle) {
+            borderColor = getThemeColor(
+              components.hpe.switch.control.track.hover.borderColor,
+              theme,
+            );
+          } else if (checked) {
+            if (toggle) {
+              borderColor = getThemeColor(
+                components.hpe.switch.control.track.selected.hover.borderColor,
+                theme,
+              );
+            } else {
+              borderColor = getThemeColor(
+                components.hpe.checkbox.control.selected.hover.borderColor,
+                theme,
+              );
+            }
+          }
+          return css`
+            ${!disabled &&
+            pad === 'none' &&
+            !toggle &&
+            `border: 2px solid ${getThemeColor(
+              components.hpe.checkbox.control.hover.borderColor,
+              theme,
+            )};`}
+            ${checked ? `border-color: ${borderColor};` : ''}
+          `;
+        }, // Q: missing token for hover borderWidth? this falls into similar boat as secondary button
+      },
+      color: components.hpe.switch.control.handle.rest.background, // The stroke color for the CheckBox icon, the toggle handle background when checked, and the border when checked. Setting to hanlde background since this is the only place to control this.
```

Comment:
```suggestion
      color: components.hpe.switch.control.handle.rest.background, // The stroke color for the CheckBox icon, the toggle handle background when checked, and the border when checked. Setting to handle background since this is the only place to control this.
```